### DISCUSSION TITLE
VirtualMin searchx frontend

### DISCUSSION THREAD

**Discussion by jebofponderworthy** [searxng/searxng#3008] [Q&A] [answered]
My server is Ubuntu 20.04.6, running current VirtualMin.  The searxng install script worked great, webapp tests AOK, I used all defaults.  I'm trying to set up a VirtualMin site to tie into it, and the documented Apache code appears to be inappropriate in a number of ways :-)  I replaced the default code (which is in a `<Directory>` block, not a `<Location>` block), and when I tried to apply settings, it gave the below.  Thoughts?  I'm definitely no apache2 expert, but I'll gladly try things, and hopefully this will help other folks using VirtualMin.

----

`Failed to apply changes : [Tue Nov 14 22:53:34.594887 2023] [so:warn] [pid 166622:tid 140399427099712] AH01574: module ssl_module is already loaded, skipping [Tue Nov 14 22:53:34.595079 2023] [so:warn] [pid 166622:tid 140399427099712] AH01574: module proxy_module is already loaded, skipping AH00526: Syntax error on line 69 of /etc/apache2/sites-enabled/search.ponderworthy.com.conf: ProxyPass cannot occur within <Directory> section Action 'configtest' failed. The Apache error log may have more information.`

**Comment by return42** [comment]
>  I replaced the default code (which is in a ``<Directory>`` block, not a ``<Location>`` block)

As the message say:

``Syntax error on line 69 of /etc/apache2/sites-enabled/search.ponderworthy.com.conf: ProxyPass cannot occur within <Directory> section Action``

> I'm definitely no apache2 expert, but I'll gladly try things,

Then the docs might be a good entry point for you: https://docs.searxng.org/admin/installation-apache.html additional https://httpd.apache.org/docs/2.4/ is a good place. There you find a link to the [mod_proxy_uwsgi](https://uwsgi-docs.readthedocs.io/en/latest/Apache.html#mod-proxy-uwsgi).

When you are looking for the meaning of a Apache directive I recommend ["Directive Index"](https://httpd.apache.org/docs/2.4/mod/directives.html).

``<Directory directory-path>``  [Description:](https://httpd.apache.org/docs/2.4/mod/core.html#directory) Enclose a group of directives that apply only to the named file-system directory, sub-directories, and their contents.

``<Location URL-path|URL>``  [Description:](https://httpd.apache.org/docs/2.4/mod/core.html#location) Applies the enclosed directives only to matching URLs

**Comment by jebofponderworthy** [comment, marked as answer]
Well, I gotta say, four thumbs up to the Searxng developers.  Have never seen anything custom go this smooth.  Here's the contents of `<Directory></Directory>` on my VirtualMin, and notice the proxy items just before.  Very, very close to the original HTTP connective code.  Works just dandy with the generic script install of the engine.

```
ProxyPreserveHost On
ProxyPass /usr/local/searxng http://127.0.0.1:8888

<Directory /home/search/public_html>

    Require all granted
    Order deny,allow
    Deny from all
    # Allow from fd00::/8 192.168.0.0/16 fe80::/10 127.0.0.0/8 ::1
    Allow from all

    # redirect / to /searxng/
    Redirect "/" "/searxng/"

    # add the trailing slash
    RedirectMatch  308 /searxng$ /searxng/

    # see flaskfix.py
    RequestHeader set X-Scheme %{REQUEST_SCHEME}s
    RequestHeader set X-Script-Name /searxng

    # see limiter.py
    RequestHeader set X-Real-IP %{REMOTE_ADDR}s
    RequestHeader append X-Forwarded-For %{REMOTE_ADDR}s

</Directory>

```
